app/src/main/java/com/hadiyarajesh/notex/reminder/MainNotificationWorker.kt

```diff
@@ -0,0 +1,31 @@
+package com.hadiyarajesh.notex.reminder
+
+import android.content.Context
+import com.hadiyarajesh.notex.database.entity.Reminder
+import java.time.Instant
+
+interface MainNotificationWorker {
+
+
+    fun createWorkRequestAndEnqueue(
+        context: Context,
+        reminder: Reminder?,
+        requestType: RequestType = RequestType.OneTimeRequest,
+        isFirstTime: Boolean = true
+    )
+
+    fun postponeRequestAndEnqueue(
+        context: Context,
+        reminderId: Long,
+        time: Instant
+    )
+
+    fun cancelWorkRequest(context: Context, tag: String): Boolean
```

Comment:
where is the required return value of this function?

and if it is not required removed Boolean from the return type.

Reply:
@sumanabhi In this case I have a doubt like where ever we trying to cancel request should need acknowledgement of cancellation, for this i put that but can remove this if there's no need. 
I have added comment for implementation also is it right way to do it or not to return boolean, was not sure regarding this.